anchore_engine/services/policy_engine/engine/feeds/config.py

```diff
@@ -0,0 +1,284 @@
+        for feed_name, sync_config_dict in data_config.items():
+            # get the default config for both verifying feed_name is a valid data feed and accessing url if necessary
+            default_sync_config = DEFAULT_PROVIDER_SYNC_CONFIG_MAP.get(provider).get(
+                feed_name
+            )
+            if default_sync_config:
+                if sync_config_dict.get("enabled", True):
+                    sync_url = sync_config_dict.get("url")
+                    if not sync_url:
+                        sync_url = default_sync_config.url
+                    to_be_synced[feed_name] = SyncConfig(
+                        enabled=True,
+                        url=sync_url,
+                        connection_timeout_seconds=connection_timeout_seconds,
+                        read_timeout_seconds=read_timeout_seconds,
+                    )
+                else:
+                    logger.debug("%s data feed is not enabled, skipping", feed_name)
+            else:  # implies feed is either invalid or unsupported for the provider
+                logger.warn(
+                    "%s data feed is unrecognized for %s provider, skipping",
+                    feed_name,
+                    provider,
+                )
+                continue
+
+    else:
+        logger.debug(
+            "Configuration found for data feeds not found, falling back to %s provider defaults",
+            provider,
+        )
+        to_be_synced = {
+            feed_name: sync_config
+            for feed_name, sync_config in DEFAULT_PROVIDER_SYNC_CONFIG_MAP.get(
+                provider
+            ).items()
+            if sync_config.enabled
+        }
+
+    logger.info(
+        "Data feeds to be synced for %s provider: %s",
+        provider,
+        list(to_be_synced.keys()),
+    )
+
+    return to_be_synced
+
+
+def is_sync_enabled(vulnerabilities_config: Dict) -> bool:
+    """
+    Returns whether a feeds sync is enabled based on the provider config. Returns true if config sections are missing
+    to be backwards compatible
+
+    :param vulnerabilities_config: vulnerabilities section of config file
+    {
+       "provider": "legacy",
+       "sync": {
+          "enabled": True,
+          "ssl_verify": True,
+          "connection_timeout_seconds": 3,
+          "read_timeout_seconds": 60,
+          "data": {
+             "vulnerabilities": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "nvdv2": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "github": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             }
+          }
+       }
+    }
+    :return: True or False
+    """
+    if not vulnerabilities_config:
+        return True
+
+    sync_enabled = vulnerabilities_config.get("sync", {}).get("enabled")
```

Comment:
can't you use `.get("enabled", True)` here and get rid of the if/else?

Reply:
That wouldn't work for the case if `enabled=""` Empty string throws off the get with defaults. If sync_enabled is blank, it would technically evaluate to False without the if-else check